### PR TITLE
feat: add pipeline.register

### PR DESCRIPTION
**Pull request by marcusschiesser**

### CODE REVIEW

File: packages/core/src/cloud/config.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  transformations?: TransformComponent[];
+  inputNodes?: BaseNode[];
+};
+
+function getTransformationConfig(
+  transformation: TransformComponent,
+): PlatformApi.ConfiguredTransformationItem {
+  if (transformation instanceof SimpleNodeParser) {
+    return {
+      configurableTransformationType: "SIMPLE_FILE_NODE_PARSER",
```
Comment (marcusschiesser) [inline packages/core/src/cloud/config.ts:19]: @himself65 have to use a string constant here as using the value from `PlatformApi.ConfigurableTransformationNames` would require a non-type import of `PlatformApi` which would be converted to a `require` call - any idea how to use this in a type safe manner with our current config?

File: packages/core/src/cloud/config.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  if (transformation instanceof OpenAIEmbedding) {
+    return {
+      configurableTransformationType: "OPENAI_EMBEDDING",
+      component: {
+        modelName: transformation.model,
+        embedBatchSize: transformation.embedBatchSize,
+        dimensions: transformation.dimensions,
+      },
+    };
+  }
```
Comment (marcusschiesser) [inline packages/core/src/cloud/config.ts:35]: @yisding @himself65 @EmanuelCampos, any alternative ideas for converting a transformation to its cloud config?

We could use a generic approach using `JSON.stringify` but then we might 
1. need to clean the serialized JSON afterwards AND
2. we might want to send parameters of sub-components to the cloud (e.g. chunksize of the textsplitter) which would be missing
- that's why I didn't do this

Adding a `toConfig` function to `TransformerComponent` is an alternative, but might be too much hassle.

File: packages/core/src/cloud/config.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+function getTransformationConfig(
+  transformation: TransformComponent,
+): PlatformApi.ConfiguredTransformationItem {
+  if (transformation instanceof SimpleNodeParser) {
+    return {
+      configurableTransformationType: "SENTENCE_AWARE_NODE_PARSER",
+      component: {
+        // TODO: API returns 422 if these parameters are included
```
Comment (marcusschiesser) [inline packages/core/src/cloud/config.ts:21]: @yisding this parameters are not used by the API

File: packages/core/src/cloud/config.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        // includePrevNextRel: transformation.includePrevNextRel,
+      },
+    };
+  }
+  if (transformation instanceof OpenAIEmbedding) {
+    return {
+      configurableTransformationType: "OPENAI_EMBEDDING",
+      component: {
+        modelName: transformation.model,
+        apiKey: transformation.apiKey,
```
Comment (marcusschiesser) [inline packages/core/src/cloud/config.ts:34]: @yisding apiKey is not used by the API

**Comment by yisding** [comment]
@marcusschiesser sorry for dropping the ball here. I'm going to merge this but we should get all of the other transformations in sooner than later.

**Pull request merged**